media_service/iiif.py

```diff
@@ -208,6 +208,11 @@ def __init__(self, manifest, sequence_id):
         self.canvases = []
 
     def add_canvas(self, canvas_id):
+        # Mirador relies on every canvas having a unique URI, so in case we have duplicate image
+        # objects in the manifest (i.e. duplicate canvases), we'll add an index to each canvas_id
+        # to ensure uniqueness within the manifest.
+        index = len(self.canvases)
+        canvas_id = "%s.%s" % (canvas_id, index)
```

Comment:
In that case, I don't think there will be a name clash. Let me walk through your example:

1. image `a` added => `[a0]`
2. image `b` added => `[a0, b1]`
3. image `c` added => `[a0, b1, c2]`
4. image `a` removed => `[b0,c1]`
5. image `b` added again => `[b0, c1, b2]`
6. imshr `a` added again => `[b0, c1, b2, a3]`

At each step or state of the collection, the manifest that is rendered and provided to Mirador should have unique canvas IDs. I should check with @rsinghal if there are any negative implications for this scheme that I'm not thinking about.